salt/states/pip_state.py

```diff
@@ -144,11 +150,7 @@ def __virtual__():
     """
     Only load if the pip module is available in __salt__
     """
-    if HAS_PKG_RESOURCES is False:
-        return False, "The pkg_resources python library is not installed"
-    if "pip.list" in __salt__:
-        return __virtualname__
-    return False
+    return "pip.list" in __salt__ and __virtualname__ or False
```

Comment:
We prefer explicit failures here... something like this:
```
if not HAS_PKG_RESOURCES:
    return False, "The pkg_resources python library is not installed"
if not "pip.list" in __salt__:
    return False, "pip.list is not available"
return __virtualname__
```
Only return the virtualname if all conditions are met. When conditions aren't met, return False, and why it was not met.

Reply:
Thanks @twangboy for the feedback! Fixed in 328b457.